sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -933,25 +941,35 @@ public Query apply(Function parent, Function inner, Context context) {
                         break;
                     case LteOperator.NAME:
                         includeUpper = true;
+                        includeLower = true;
                         to = distance;
                         break;
                     case LtOperator.NAME:
                         to = distance;
+                        includeLower = true;
                         break;
                     case GteOperator.NAME:
                         from = distance;
                         includeLower = true;
+                        includeUpper = true;
                         break;
                     case GtOperator.NAME:
                         from = distance;
+                        includeUpper = true;
                         break;
                     default:
                         // invalid operator? give up
                         return null;
                 }
-                GeoPoint geoPoint = new GeoPoint(lat, lon);
 
-                return new GeoDistanceRangeQuery(
+                final Version indexCreated = Version.indexCreated(context.indexCache.indexSettings());
+                if (indexCreated.onOrAfter(Version.V_2_2_0)) {
+                    GeoUtils.normalizePoint(geoPoint);
+                }
+
+                final Query query;
+                if(indexCreated.before(Version.V_2_2_0)) {
```

Comment:
please only check for V_2_3_0 like in `WITHIN` https://github.com/crate/crate/blob/geo-improvements-es2.2/sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java#L841